js/app.js

```diff
@@ -0,0 +1,159 @@
+const app = {
+
+    appName: 'Island Race',
+    version: '1.0.0',
+    license: undefined,
+    author: 'Alejandro Abad',
+    description: 'Racing Cars Game',
+    ctx: undefined,
+    canvasSize: {
+        w: undefined, h: undefined
+    },
+
+    carData: {
+        pos: { x: this.canvasSize.w / 2, y: this.canvasSize.h },
```

Comment:
No puedes definir la posicion asi, porque cuando se ejecuta esta linea this.canvasSize es undefined por lo que como este valor no se actuliza es siempre undefined. Te recomiendo que la posicion inicial del coche la pongas a mano